src/test/java/com/palantir/docker/compose/DockerCompositionIntegrationTest.java

```diff
@@ -28,10 +29,17 @@
 
 public class DockerCompositionIntegrationTest {
 
+    private final DockerService postgresFromComposeFileSnippet = DockerService.fromDockerCompositionFile("src/test/resources/postgres-service.yaml");
+    private final DockerService postgresDefinedInline = DockerService.fromImage("kiasaki/alpine-postgres", "inlinePostgres")
+                                                                     .withPortMapping(5432)
+                                                                     .build();
+
     @Rule
     public DockerComposition composition = DockerComposition.of("src/test/resources/docker-compose.yaml")
                                                             .waitingForService("db", toHaveAllPortsOpen())
                                                             .waitingForService("db2", toHaveAllPortsOpen())
+                                                            .withService(postgresFromComposeFileSnippet)
```

Comment:
Looking at how `DockerComposition` is now going to be used, it seems a bit weird that you have a composition `of` some yaml file, which could be whatever you want, and then you have the concept of a service, which is also from a yaml file.

IMO it would make more sense if the file parameter in `of` was gotten rid of, and instead it would consist of a varArgs list of `DockerService`'s.